Interfaces pour les différentes pages

HiraAkshayy opened the pull request.

MasterPNJ committed:
Create Menu.java

First commit !

MasterPNJ committed:
Update .gitignore

Update gitignore

MasterPNJ committed:
Update Menu.java

MasterPNJ committed:
Update Menu.java

Update menu redimensionnement

MasterPNJ committed:
Image/Icon de jeu

Mise en place des différents logo/image actuel

MasterPNJ committed:
Update menu/image

Update menu ainsi que l'ajout des images concernant le titre

MasterPNJ committed:
Update Menu.java

Ajout de listener pour changer la couleur des boutons lorsque la souris passe au dessus.
Fix plusieurs lignes.

tetelie committed:
ajout du Menu du mode libre

tetelie committed:
Merge remote-tracking branch 'origin/Interface' into Interface

Commit:
Ajout du menu de démarrage

HiraAkshayy committed:
Création du menu de techniques (unfinished)

Commit:
Merge branch 'Interface' of https://github.com/projet-gl2/Hashimiste into Interface
Mise a jour de la branche

HiraAkshayy committed:
Merge branch 'Interface' of https://github.com/projet-gl2/Hashimiste

MasterPNJ committed:
Update Menu.java

Remplacement des magiques numbers par des constantes.
Fix du redimensionnement de la fenêtre et des boutons.
Add des commentaires.

MasterPNJ committed:
Update Menu.java

Fix de certains warnings

tetelie committed:
update ModeLibre.java + doc

tetelie committed:
Merge branch 'Interface' of github.com:projet-gl2/Hashimiste into Interface

HiraAkshayy committed:
Création du menu des paramètres et légères modifications sur le menu des techniques

tetelie committed:
ajout du Menu du mode libre

HiraAkshayy committed:
Merge branch 'Interface' of https://github.com/projet-gl2/Hashimiste into Interface

tetelie committed:
Merge branch 'master' into Interface

HiraAkshayy committed:
Update MenuTechnique - Ajout de javadoc

MasterPNJ committed:
Update Menu.java

Ajout de la couleur de fond de la fenêtre.
Appel au menu paramètre.
Correction des "é" en "e" pour la javadoc.

HiraAkshayy committed:
Update MenuParametres - Ajout javadoc

HiraAkshayy committed:
Merge remote-tracking branch 'origin/master' into Interface

HiraAkshayy committed:
Merge branch 'master' of https://github.com/projet-gl2/Hashimiste into Interface

MasterPNJ committed:
Ajout de Couleur.java + Maj de Menu.java

Création de Couleur.java pour gérer les différentes couleurs nécessaires.
Mise à jour de Menu.java en utilisant les méthodes de Couleur.java.

MasterPNJ committed:
Ajout de image.java + maj de menu.java + maj de la doc

Ajout de Image.java.
Maj de la doc.
Maj de menu.

tetelie committed:
Update Menu Libre

correction affichage + compatibilité avec les maps

tetelie committed:
Merge branch 'Interface' of github.com:projet-gl2/Hashimiste into Interface

HiraAkshayy committed:
Update  - implémentation de l'enum Technique

HiraAkshayy committed:
Merge branch 'Interface' of https://github.com/projet-gl2/Hashimiste into Interface

MasterPNJ committed:
Update Menu.java

MasterPNJ committed:
Merge branch 'Interface' of https://github.com/projet-gl2/Hashimiste into Interface

HiraAkshayy committed:
Merge branch 'master' of https://github.com/projet-gl2/Hashimiste into Interface

HiraAkshayy committed:
Update - gestion couleurs avec classe Couleur

tetelie committed:
Ajout des couleurs + modif du look and feel pour mac

Ajouts des couleurs avec le thème principale stocké dans Couleur.java
Modification du look and feel pour une meilleure compatibilité des couleurs sur macOs

tetelie committed:
Merge branch 'Interface' of github.com:projet-gl2/Hashimiste into Interface

HiraAkshayy committed:
Ajout d'un package pour l'importation

HiraAkshayy committed:
Update - changement chemin package

HiraAkshayy committed:
Merge branch 'Interface' of https://github.com/projet-gl2/Hashimiste into Interface

HiraAkshayy committed:
Suppression de fichier

Commit:
ajout transparence mise a jour menuprofilcreation

Commit:
Merge branch 'Interface' of https://github.com/projet-gl2/Hashimiste into Interface

enimaloc commented on src/main/java/fr/hashimiste/gui/Couleur.java:
```diff
@@ -0,0 +1,76 @@
+    private static java.awt.Color couleurTextBouton = new java.awt.Color(251, 250, 242);
+    private static java.awt.Color couleurBoutonDesactive = new java.awt.Color(197, 179, 179);
+    private static java.awt.Color couleurFond = new java.awt.Color(251,250,242);
+    public final static java.awt.Color couleurTransparent=new java.awt.Color(0x00000000, true);
```
Import ces classes directement dans le headers du fichier

```java
import java.awt.Color;
```

enimaloc commented on src/main/java/fr/hashimiste/gui/Couleur.java:
```diff
@@ -0,0 +1,76 @@
+     */
+    public void setCouleurFond(java.awt.Color couleurFond) {
+        Couleur.couleurFond = couleurFond;
+    }
```
Ici, cette classe représente une classe de constante je ferais donc plus

```suggestion
    public final static Color COULEUR_BOUTON = new Color(160, 158, 188);
    public final static Color COULEUR_TEXTE_BOUTON = new Color(251, 250, 242);
    public final static Color COULEUR_BOUTON_DESACTIVE = new Color(197, 179, 179);
    public final static Color COULEUR_FOND = new Color(251,250,242);
    public final static Color COULEUR_TRANSPARENT = new Color(0x00000000, true);
    
    /**
      * Constructeur privée pour éviter l'instanciation
     **/
    private Couleur() {}
```

Sans oublié la documentation

enimaloc commented on src/main/java/fr/hashimiste/gui/Image.java:
```diff
@@ -0,0 +1,63 @@
+    private final static String logo = "src/main/resources/images/logo.png";
+    private final static String logoOriginal = "src/main/resources/images/logoOriginal.png";
+    private final static String logoTitre = "src/main/resources/images/logoTitre.png";
+    private final static String logoTitreTransparent = "src/main/resources/images/logoTitreTransparent.png";
```
La convention pour les constantes est `UPPER_SNAKE_CASE` est non en `camelCase`

enimaloc commented on src/main/java/fr/hashimiste/gui/Menu.java:
```diff
@@ -0,0 +1,476 @@
+    private final JButton ButParametre = new JButton("Parametres");
+    private final JButton ButProfils = new JButton("Profils");
+    private final JLabel lblSomeText = new JLabel("Hashimiste", SwingConstants.CENTER);
+    private final int tailleMinX = 500;
+    private final int tailleMinY = 300;
+    private final int taillePrefX = 550;
+    private final int taillePrefY = 350;
+    private final int tailleMaxX = 1920;
+    private final int tailleMaxY = 1080;
+    private final int espace = 10;
+    private final float coefWidth1 = 0.25f;
+    private final float coefHeight1 = 0.08f;
+    private final float coefWidth2 = 0.1f;
+    private final float coefHeight2 = 0.05f;
+    private final int tailleButParamProfilX = 100;
+    private final int tailleButY = 20;
+    private final int tailleButX = 50;
```
Les constantes sont en `UPPER_SNAKE_CASE` et non en `camelCase`

enimaloc replied:
Maybe mettre en `static` cela ne sert à rien de recréer une variable pour ces constante primitive

DamiBoul replied:
je veux pas être rabat joie mais il me semble qu'il faut aussi mettre de la documentation sur les constantes ?

enimaloc commented on src/main/java/fr/hashimiste/gui/Menu.java:
```diff
@@ -0,0 +1,476 @@
+        jp.setBackground(new java.awt.Color(Couleur.getCouleurFond().getRGB()));
+        jp2.setBackground(new java.awt.Color(Couleur.getCouleurFond().getRGB()));
+        this.setVisible(true);
+    }
```
Réduire cette méthode d'URGENCE, 350 lignes ce n'est pas possible !

> Ce point est un veto pour l'approbation de la PR

Utilise l'import de classe tel que

```java
import java.awt.Dimension;
import java.awt.Color;
import java.awt.event.MouseListener;
import java.awt.event.MouseEvent;
```

enimaloc commented on src/main/java/fr/hashimiste/gui/MenuParametres.java:
```diff
@@ -0,0 +1,108 @@
+    private JPanel jp = new JPanel();
+    private JButton butMenu = new JButton("Menu");
+
+    private Color couleurBouton = new java.awt.Color(160, 158, 188);
+    private Color couleurTextBouton = new java.awt.Color(251, 250, 242);
```
On peut enlever `java.awt.` car la classe est déjà importé par `import java.awt.*;`

enimaloc commented on src/main/java/fr/hashimiste/gui/MenuParametres.java:
```diff
@@ -0,0 +1,108 @@
+    private Color couleurBouton = new java.awt.Color(160, 158, 188);
+    private Color couleurTextBouton = new java.awt.Color(251, 250, 242);
+
+    private int tailleMinX = 500;
+    private int tailleMinY = 300;
+    private int tailleMaxX = 1920;
+    private int tailleMaxY = 1080;
```
Peut être des constantes

enimaloc commented on src/main/java/fr/hashimiste/gui/Image.java:
```diff
@@ -0,0 +1,63 @@
+        return logoTitreTransparent;
+    }
+
+}
```
Cette classe est une classe constante on peut la remplacer par:

```suggestion
public class Image {
    public final static String cheminIcone = "src/main/resources/images/icon.png";
    public final static String cheminIconeTransparent = "src/main/resources/images/iconTransparent.png";
    public final static String logo = "src/main/resources/images/logo.png";
    public final static String logoOriginal = "src/main/resources/images/logoOriginal.png";
    public final static String logoTitre = "src/main/resources/images/logoTitre.png";
    public final static String logoTitreTransparent = "src/main/resources/images/logoTitreTransparent.png";
    
    /**
      * Constructeur privée pour éviter l'instanciation
     **/
    private Image() {}
}
```

Sans oublier la documentation

enimaloc commented on src/main/java/fr/hashimiste/gui/MenuProfilCreation.java:
```diff
@@ -0,0 +1,115 @@
+*/
+public class MenuProfilCreation extends JFrame {
+   
+    /** Constructeur de la classe MenuProfilCreation permet de créer une interface graphique. 
+     * 
+    */
```
La javadoc ne correspond pas a ce qu'elle cible

enimaloc commented on src/main/java/fr/hashimiste/gui/MenuProfilCreation.java:
```diff
@@ -0,0 +1,115 @@
+        //}
+
+        this.setSize(500, 300);
+        this.setMinimumSize(new java.awt.Dimension(1000, 720));
```
Import ces classes directement dans le headers du fichier

```java
import java.awt.Dimension;
```

enimaloc commented on src/main/java/fr/hashimiste/gui/MenuProfilCreation.java:
```diff
@@ -0,0 +1,115 @@
+        
+        return tmp;
+    }
+   public static void main(String[] args){
```
Manque de la javadoc ici

enimaloc commented on src/main/java/fr/hashimiste/gui/MenuTechnique.java:
```diff
@@ -0,0 +1,153 @@
+
+    private ArrayList<String> listDesc = new ArrayList<>();
+
+    private int tailleMinX = 950;
+    private int tailleMinY = 400;
+    private int tailleMaxX = 1920;
+    private int tailleMaxY = 1080;
```
Peut être déclarer en tant que constantes

enimaloc commented on src/main/java/fr/hashimiste/gui/modelibre/ModeLibre.java:
```diff
@@ -0,0 +1,212 @@
+
+    private final PreviewComponent bigPreview = new PreviewComponent(Color.GRAY, grille);
+
+    private final int tailleMinX = 550;
+    private final int tailleMinY = 500;
+    private final int tailleMaxX = 1920;
+    private final int tailleMaxY = 1080;
```
Peut être mis en `static`

enimaloc commented on src/main/java/fr/hashimiste/gui/modelibre/ModeLibre.java:
```diff
@@ -0,0 +1,212 @@
+                System.out.println("bip");
+                Point point = new Point(e.getX(), e.getY());
+                Component c = panelFacile.getComponentAt(e.getPoint());
+                System.out.println(c.getClass());
```
Ne pas oublier de retirer les variable de débug

enimaloc commented on src/main/java/fr/hashimiste/gui/modelibre/PreviewComponent.java:
```diff
@@ -0,0 +1,66 @@
+                g.setColor(Color.black);
+                int t = d / 7;
+                g.drawOval(x + t * ile.getX(), y + t *ile.getY(), t, t);
+                g.setFont(new Font("Andale Mono", Font.BOLD, t));
```
Déclarer l'instance en tant que constante sera bénéfique

enimaloc commented on src/main/java/fr/hashimiste/gui/modelibre/PreviewComponent.java:
```diff
@@ -0,0 +1,66 @@
+    {
+        this.grille = grille;
+        this.color = color;
+        this.setMinimumSize(new Dimension(50,50));
```
Déclarer les Magic Value en tant que constante sera bénéfique

enimaloc commented on src/main/java/fr/hashimiste/gui/modelibre/PreviewComponent.java:
```diff
@@ -0,0 +1,66 @@
+import java.awt.event.MouseEvent;
+import java.util.Random;
+
+public class PreviewComponent extends JComponent {
```
Manque de la javadoc

enimaloc commented on src/main/java/fr/hashimiste/gui/modelibre/PreviewComponent.java:
```diff
@@ -0,0 +1,66 @@
+        return color;
+    }
+
+    public void setColor(Color color) {
```
Manque de la javadoc

enimaloc commented on src/main/java/fr/hashimiste/gui/modelibre/PreviewComponent.java:
```diff
@@ -0,0 +1,66 @@
+
+    }
+
+    public Color getColor() {
```
Manque de la javadoc

enimaloc commented on src/main/java/fr/hashimiste/techniques/Ile.java:
```diff
@@ -0,0 +1,94 @@
+    private int y; //coordonée en colonnes
+    private Grille grille; //grille dans laquelle se trouve l'île
+
+    public int getN() {
```
Manque de la javadoc

enimaloc commented on src/main/java/fr/hashimiste/techniques/Ile.java:
```diff
@@ -0,0 +1,94 @@
+        return n;
+    }
+
+    public int getX() {
```
Manque de la javadoc

enimaloc commented on src/main/java/fr/hashimiste/techniques/Ile.java:
```diff
@@ -0,0 +1,94 @@
+        return x;
+    }
+
+    public int getY() {
```
Manque de la javadoc

enimaloc commented on src/main/java/fr/hashimiste/techniques/Ile.java:
```diff
@@ -0,0 +1,94 @@
+        return y;
+    }
+
+    public Grille getGrille() {
```
Manque de la javadoc

enimaloc commented on src/main/java/fr/hashimiste/techniques/Ile.java:
Il faudrais récupérer l'état actuelle de la branche pour pouvoir remplacer cette classe par `fr.hashimiste.maps.Ile`

enimaloc commented on src/test/java/fr/hashimiste/gui/ModeLibreGui.java:
```diff
@@ -0,0 +1,23 @@
+
+public class ModeLibreGui {
+
+    public static void main(String[] args) {
```
Manque de la javadoc

enimaloc commented on src/test/java/fr/hashimiste/gui/ModeLibreGui.java:
```diff
@@ -0,0 +1,23 @@
+
+import javax.swing.*;
+
+public class ModeLibreGui {
```
Manque de la javadoc

enimaloc commented on src/test/java/fr/hashimiste/gui/ModeLibreGui.java:
Je suppose que cette classe est toujours en cours de construction, indiquer là avec `#TODO: A implémenter`

tetelie replied:
c'est une classe de test

enimaloc commented on src/main/resources/images/iconTransparent.png:
Au lieu de mettre plusieurs image, peut-être modifier une image originale en plusieurs variantes ?

enimaloc commented on src/main/resources/images/icon.png:
Au lieu de mettre plusieurs image, peut-être modifier une image originale en plusieurs variantes ?

enimaloc commented on src/main/resources/images/logo.png:
Au lieu de mettre plusieurs image, peut-être modifier une image originale en plusieurs variantes ?

enimaloc commented on src/main/resources/images/logoTitre.png:
Au lieu de mettre plusieurs image, peut-être modifier une image originale en plusieurs variantes ?

enimaloc commented on src/main/resources/images/logoTitreTransparent.png:
Au lieu de mettre plusieurs image, peut-être modifier une image originale en plusieurs variantes ?

MasterPNJ committed:
Modification de couleur, image et menu.java par rapport au retour de la review

Ajout de la doc.
Respect des conventions de nommages.
Allègement du constructeur de menu avec l'ajout de plusieurs méthodes.

tetelie committed:
javadoc + todo

tetelie committed:
Merge branch 'Interface' of github.com:projet-gl2/Hashimiste into Interface

tetelie committed:
modification des couleurs
